src/app/time-entry/time-entry.component.spec.ts

```diff
@@ -69,12 +69,15 @@ describe('TimeEntryComponent', () => {
     const timeEntryElement = fixture.debugElement;
     const buttonElement = timeEntryElement.query(By.css(('.input-group-text')));
     const toggleLogPeriodSpy = spyOn(component, 'toggleLogPeriod').and.callThrough();
+    const periodInputSelector = '[formcontrolname="period"]';
```

Comment:
1. When testing toggle, I would expect check not only for `[formcontrolname="period"]` but also for `[formcontrolname="from"]`.  Please add, as it was reported as a bug.
2. The test checks toggle action, so the description 'have button' is not acurate